### PR TITLE
API maintenance

### PR DESCRIPTION
**Pull request by biquando**
This PR cleans up the backend API code in preparation for integration with the bot and frontend.

**Important changes:**
- Generate static documentation in `/documentation/api.html`
- Change root directory to be `DiscordRC/` instead of `DiscordRC/api/` (*This changes how to start the backend. See `api/README.md` for updated instructions.*)
- Simplify User model by storing only a list of server names, rather than a mapping from server names to ids. If you had any test users in your database, then you will probably have to delete & recreate them.

### CODE REVIEW

File: api/README.md [hunk 1]
```diff
@@ -4,6 +4,6 @@
 2. Activate virtual environment: `. venv/bin/activate` (if you're using bash)
 3. Install dependencies: `pip3 install -r requirements.txt`
 4. Set up database in docker: `./mongo_setup.sh`
-5. Run: `uvicorn main:app --reload`
+5. Run: `sh -c 'cd .. && uvicorn api.main:app --reload'`
```
Comment (biquando) [inline api/README.md:7]: This change is important!!! Make sure to use this new command whenever you run the backend.

File: api/models.py [hunk 2]
```diff
@@ -73,20 +77,12 @@ class TokenData(BaseModel):
...
-
 class User(BaseModel):
     '''
-    Data about a user. `servers` maps `ServerConfigInDB.name` to
-    `ServerConfigInDB.uid`.
+    Data about a user. `servers` is a list of server names.
     '''
     username: str
-    servers: list[UserConfigMapping] = list()
+    servers: list[str] = list()
```
Comment (biquando) [inline api/models.py:85]: This change is important!!! It simplifies how a User stores its server configs.

File: api/routes/bot.py
Comment (biquando) [inline api/routes/bot.py:1]: These endpoints are WIP

File: api/documentation.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    <script src="https://cdn.jsdelivr.net/npm/redoc/bundles/redoc.standalone.js"> </script>
+    <script>
+        var spec = %s;
+        Redoc.init(spec, {}, document.getElementById("redoc-container"));
+    </script>
+</body>
+</html>
+'''
+
+def generate(app, filepath):
```
Comment (sdcarlson) [inline api/documentation.py:34]: add a UT for this

File: api/models.py [hunk 4]
```diff
@@ -18,17 +22,17 @@ class Channel(BaseModel):
     '''
     name: str
     id: int | None = None
-    permissions: dict[str, dict[str, bool]] = dict()
+    permissions: dict[str, dict[str, bool]] = {}
```
Comment (sdcarlson) [inline api/models.py:25]: add UTs to make sure all these new models are behaving as expected

**Review by sdcarlson — CHANGES_REQUESTED** [review]
Looks good to me but you should add some more UTs please

File: api/models.py (hunk 4 above)
Comment (biquando) [reply api/models.py:25]: I'll add UTs in another PR

**Review by kevwjin — APPROVED** [review]
LGTM

**Pull request merged**